java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java

```diff
@@ -130,7 +130,7 @@ private void handleSuppressWarning(List<AnnotationTree> annotationTrees, Tree tr
     List<String> rules = Lists.newArrayList();
     for (AnnotationTree annotationTree : annotationTrees) {
       if (isSuppressWarningsAnnotation(annotationTree)) {
-        startLine = ((JavaTree) annotationTree).getLine();
+        startLine = startLineIncludingTrivia(tree);
```

Comment:
Indeed, I miss-read the code. Sorry!
To me we were only excluding the lines from the trivia attached to that the very first token, and not selecting the highest line as start-line for the whole exclusion.